apps/web/src/views/PositionManagers/hooks/useFilters.ts

```diff
@@ -7,17 +7,21 @@ function filterHookFactory(filterName: string) {
     const router = useRouter()
     const filter = useMemo(() => {
       const value = router.query[filterName]
-      if (Array.isArray(value)) {
-        return value[0]
+      if (value) {
+        if (Array.isArray(value)) {
+          return decodeURIComponent(value[0])
+        }
+        return decodeURIComponent(value)
       }
-      return value
+      return undefined
     }, [router.query])
 
     const setFilter = useCallback(
       (value?: string) => {
+        const trimmedValue = value?.trim()
         router.push(
           updateQuery(router.asPath, {
-            [filterName]: value,
+            [filterName]: trimmedValue && encodeURIComponent(trimmedValue),
```

Comment:
I don't think we need to encode and decode again? `updateQuery` is using `url-parse` to parse and serialize the url, which uses `querystringify` to encode and decode the querystring

https://github.com/pancakeswap/pancake-frontend/blob/develop/packages/utils/clientRouter.ts#L1-L17

Reply:
May I ask what's the bad case here?